app/src/main/java/org/simple/clinic/editpatient/EditPatientModel.kt

```diff
@@ -57,8 +57,8 @@ data class EditPatientModel(
   fun updateDateOfBirth(dateOfBirth: String): EditPatientModel =
       copy(ongoingEntry = ongoingEntry.updateDateOfBirth(dateOfBirth))
 
-  fun updateBangladeshNationalId(bangladeshNationalId: BusinessId): EditPatientModel =
-      copy(ongoingEntry = ongoingEntry.updateBangladeshNationalId(bangladeshNationalId))
+  fun updateSavedBangladeshNationalId(bangladeshNationalId: BusinessId): EditPatientModel =
+      copy(savedEntry = savedEntry.updateBangladeshNationalId(bangladeshNationalId))
```

Comment:
Yea. I figured this too. But it would have needed more time.
I will create a story for this refactoring. I was also considering replacing saved states for individual fields with this `savedEntry` object. Will see which is better when we pick this up.

Reply:
Added story-> https://www.pivotaltracker.com/story/show/171175597